github-issue/check.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+import requests
+import json
+import sys
+import os
+import datetime
+from pprint import pprint
+import dateutil.parser
+
+def _check(instream):
+    payload = json.load(instream)
+    source = payload['source']
+    token = source['token']
+    repository = source['repository']
+    try:
+        version = [payload['version']]
+    except (KeyError,NameError):
+        version = []
+
+    headers = {'Authorization': 'token ' + token}
+
+    # first get the last page if it exists
+    endpoint = "https://api.github.com/repos/" + repository + "/issues?sort=updated&direction=asc"
+    connection = requests.get(endpoint, headers=headers)
+    link = connection.headers.get('link')
+    if link is not None:
+        links = link.split(',')
+        for link in links:
+            if 'rel="last"' in link:
```

Comment:
if the last page just happens to only have like 2 things on it, that would kind of suck. if you sort by DESC, only use the first page, and then reverse the results before returning them, at least you'll at a minimum get a full page worth of stuff